### PR TITLE
[DEV APPROVED] 7709 internal links snippet

### PR DESCRIPTION
**Pull request by jriga**
**Feature**
[Card 7709](https://moneyadviceservice.tpondemand.com/entity/7709)

**Solution**
Added a internal link post processor to generate the menu after the `h1 + p` tags

### CODE REVIEW

**Comment by jongilbraith** [comment]
Some interesting patterns there. 👍

**Comment by tomas-stefano** [comment]
Besides the minor comment looks good 👍

**Comment by Tr4pSt3R** [comment]
@jriga just in case you are wondering why CodeClimate is failing

```
 () ~/Projects/MAS/cms (7709-internal-links-snippet)
⤷  flog -vd app/models/internal_link.rb
** flogging app/models/internal_link.rb
    57.6: flog total
     6.4: flog/method average

    15.2: InternalLink#list_items          app/models/internal_link.rb:30
     3.2:   to_proc_normal
     3.2:   block_pass
     2.8:   map
     1.7:   sections
     1.7:   add_id_attribute
     1.5:   to_li_tag
     1.1:   join

    15.0: InternalLink#call                app/models/internal_link.rb:14
     2.9:   doc
     2.6:   present?
     2.2:   branch
     1.6:   internal_links_menu
     1.4:   internal_links
     1.4:   sections
     1.4:   css
     1.2:   !
     1.2:   after
     1.1:   to_s

     8.3: InternalLink#initialize          app/models/internal_link.rb:6
     4.4:   assignment
     2.6:   doc
     2.2:   css
     1.1:   fragment
     1.1:   freeze
```

**Comment by Tr4pSt3R** [comment]
👍

File: spec/models/content_composer_spec.rb [hunk 1]
```diff
@@ -37,7 +37,7 @@
 
    describe '#post_processors' do
      it 'returns TableWrapper, TableCaptioner and ExternalLink' do
```
Comment (jongilbraith) [inline spec/models/content_composer_spec.rb]: Need to update the example description come to think of it.

Comment (jongilbraith) [inline spec/models/content_composer_spec.rb]: Need to update the example description come to think of it.

File: bin/rspec [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env ruby
+begin
+  load File.expand_path('../spring', __FILE__)
+rescue LoadError => e
+  raise unless e.message.include?('spring')
+end
+require 'bundler/setup'
+load Gem.bin_path('rspec-core', 'rspec')
```
Comment (jriga) [inline bin/rspec]: this is the rspec binstub for spring for faster specs running

**Pull request merged**